### PR TITLE
Add ability to provide cancel reason

### PR DESCRIPTION
**Pull request by szmarczak**

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,8 +1,8 @@
 'use strict';
 
 class CancelError extends Error {
-	constructor() {
-		super('Promise was canceled');
+	constructor(reason = '') {
+		super('Promise was canceled. Reason: ' + reason);
```
Comment (sindresorhus) [inline index.js]: It should not show the word `Reason: ` if no reason is specified.

**Comment by sindresorhus** [comment]
Yeah, probably a good idea to encourage this. Needs docs and a test.

File: index.js (hunk 1 above)
Comment (sindresorhus) [inline index.js]: Maybe the provided reason should just replace the message entirely? It's already clear from the error name that it was canceled.

**Comment by szmarczak** [comment]
Here you go :unicorn:

**Comment by sindresorhus** [comment]
🙌🦄

**Pull request merged**